### ISSUE TITLE
Testing a complex reflection group

### ISSUE THREAD

**Issue by sebastiendesignolle** [replab/replab#89]
When trying to work with the [complex reflection group](https://en.wikipedia.org/wiki/Complex_reflection_group) with Shephard-Todd number 29, the decomposition in irreducible representations fails. Here is the minimal code ; the generators were obtained with [GAP 3](https://webusers.imj-prg.fr/~jean.michel/gap3/htm/chap084.htm#SECT006).

    g1 = [5,6,3,4,1,2,10,29,19,7,14,12,13,11,17,30,15,18,9,22,31,20,25,24,23,28,32,26,8,16,21,27,75,76,35,77,37,78,79,80,81,51,43,82,83,84,48,47,50,49,42,85,86,87,88,89,90,58,91,92,93,62,94,64,95,96,67,97,98,99,100,72,101,102,33,34,36,38,39,40,41,44,45,46,52,53,54,55,56,57,59,60,61,63,65,66,68,69,70,71,73,74,147,148,105,149,125,108,126,110,112,111,118,150,115,116,117,113,119,151,121,122,123,124,107,109,152,129,128,134,153,132,133,130,135,154,138,137,155,140,141,142,156,146,145,144,103,104,106,114,120,127,131,136,139,143];
    g2 = [6,7,8,9,10,1,2,3,4,5,11,33,34,35,15,36,37,20,19,18,38,22,39,40,41,42,43,44,29,45,46,32,12,13,14,16,17,21,23,24,25,26,27,28,30,31,103,104,105,50,82,53,52,72,65,106,57,107,59,60,108,66,67,68,55,62,63,64,69,109,71,54,110,74,75,76,83,84,81,80,79,51,77,78,111,112,87,113,114,115,116,117,93,94,118,96,97,119,120,121,101,122,47,48,49,56,58,61,70,73,85,86,88,89,90,91,92,95,98,99,100,102,143,136,125,151,127,128,133,132,131,130,129,134,137,124,135,138,142,140,141,139,123,144,145,157,148,147,150,149,126,158,159,154,155,156,146,152,153];
    g3 = [1,11,12,13,5,14,15,8,16,17,2,3,4,6,7,9,10,24,30,47,42,48,27,18,32,49,23,50,29,19,51,25,33,52,35,53,37,54,55,56,57,21,58,59,60,61,20,22,26,28,31,34,36,38,39,40,41,43,44,45,46,110,63,123,107,66,124,68,70,69,109,105,73,74,75,85,86,87,88,89,90,91,92,93,76,77,78,79,80,81,82,83,84,94,125,96,97,99,98,126,101,102,106,127,72,103,65,116,71,62,128,129,130,131,132,108,133,134,119,120,121,135,64,67,95,100,104,111,112,113,114,115,117,118,122,136,137,138,139,140,141,157,143,144,145,146,149,152,147,153,151,148,150,154,155,156,142,159,158];
    g4 = [1,9,13,18,5,19,20,21,2,22,23,24,3,25,26,27,28,4,6,7,8,10,11,12,14,15,16,17,31,32,29,30,55,38,62,63,64,34,39,65,66,42,67,68,45,46,49,50,47,48,51,69,53,70,33,71,57,72,59,73,74,35,36,37,40,41,43,44,52,54,56,58,60,61,88,78,94,76,79,95,96,97,83,84,98,86,99,75,100,90,91,101,102,77,80,81,82,85,87,89,92,93,136,104,124,106,109,108,107,110,137,138,113,139,140,141,117,118,135,120,142,122,123,105,126,125,143,128,129,144,131,132,145,146,119,103,111,112,114,115,116,121,127,130,133,134,154,148,149,155,151,156,153,147,150,152,157,158,160,159];
    g1 = [g1,length(g1)+1:160];
    g2 = [g2,length(g2)+1:160];
    g3 = [g3,length(g3)+1:160];
    g4 = [g4,length(g4)+1:160];
    G = replab.Permutations(160).subgroup({g1 g2 g3 g4});
    disp(G.order) % 7680
    rho = G.naturalRep;
    rho.decomposition

**Comment by denisrosset** [comment]
Not a huge bug, it's the rational basis recovery that's too fragile.

Rational basis recovery is a huge help for pedagogical examples, but I need to check why it fails, enclose it in a `try ... catch` (it is not mandatory), and maybe restrict its use for small bases.

**Comment by denisrosset** [comment]
Workaround: in line 140 of `Rep.m`, replace

`                self.decomposition_ = dec.nice;`

by

`                self.decomposition_ = dec;`

**Comment by jdbancal** [comment]
Thanks for the quick workaround! I can confirm that the change at line 140 works.

It seems that U0rational is transposed compared to what it should be: in line 20 of `niceRep.m`, `size(sub.U0)' is 20x160, whereas `size(U0rational)` is 160x20...

**Comment by denisrosset** [comment]
I've wrapped the rational recovery in try/catch blocks for the time being, and added a test for this example.

@sebastiendesignolle Right now, I'm not doing much with the result of the decomposition. Are there references that I can base tests on? The result I get is:
```
 component(1): Isotypic component R(1)      
 component(2): Isotypic component R(4)      
 component(3): Isotypic component R(5)      
 component(4): Isotypic component C(8)      
 component(5): Isotypic component I(2)xR(10)
 component(6): Isotypic component R(15)     
 component(7): Isotypic component R(15)     
 component(8): Isotypic component R(20)     
 component(9): Isotypic component C(32)     
component(10): Isotypic component C(40)     
```

**Comment by denisrosset** [comment]
I'll close this issue when we know what to do with the test

**Comment by jdbancal** [comment]
Is it necessary to include a `try ... catch` block here? What about `U0rational` being seemingly transposed?

**Comment by denisrosset** [comment]
It's a me being lazy thing.

I plan to refactor slightly and comment the rep decomposition functions, I'll check (and document) the matrix shapes as part of that.

**Comment by jdbancal** [comment]
Ok, I let you manage that!

**Comment by denisrosset** [comment]
Original bug is corrected, but questions about the test are in #97
